src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -1357,12 +1467,14 @@ static uint16_t icap_get_memidx(struct mem_topology *mem_topo, enum IP_TYPE ecc_
 	return memidx;
 }
 
-static int icap_create_subdev_debugip(struct platform_device *pdev)
+static int icap_create_subdev_debugip(struct platform_device *pdev,
+				      uint32_t slot_id)
 {
 	struct icap *icap = platform_get_drvdata(pdev);
 	int err = 0, i = 0;
 	xdev_handle_t xdev = xocl_get_xdev(pdev);
-	struct debug_ip_layout *debug_ip_layout = icap->debug_layout;
+	struct debug_ip_layout *debug_ip_layout =
+		icap->slot_info[slot_id]->debug_layout;
```

Comment:
Can slot_info[slot_id] be NULL in this function? Just like many other functions.

Reply:
Fixed. Thanks